flink-runtime/src/main/java/org/apache/flink/runtime/taskmanager/Task.java

```diff
@@ -914,6 +914,11 @@ private void closeNetworkResources() {
 				LOG.error("Failed to release input gate for task {}.", taskNameWithSubtask, t);
 			}
 		}
+		try {
+			taskStateManager.close();
```

Comment:
The precious consideration of closing network resources early is done by canceler thread, which would throw exception while task thread interacts with buffer operation to make task exit ASAP. If we add the state manager close here, does it have the same effect to throw any exceptions while interacting with task thread?

Reply:
I don't fully understand your concerns, could you explain what are they?

On close, `taskStateManager` will close all opened input streams (without waiting for the task thread).

Reply:
The previous `closeNetworkResources` can be used in two scenarios, one is for task exit in finally region, another is used by canceler thread before task exiting. The motivation to close network resource by canceler thread is to release the buffers ASAP, then the task thread can cause exception while interacting with buffer to make it exit early.

If the close of `taskStateManager` does not have the effect to make task thread exit early, then it should not be called by canceler thread, and it only needs to be called in task thread finally region.

Reply:
Thanks for the explanation.
I will move `taskStateManager.close()` to `releaseResources()` which is called from `doRun` and not task canceller.